language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/SubmitCommandsRequest.java

```diff
@@ -261,6 +261,87 @@ private static CommandsOuterClass.Commands toProto(
     return builder.build();
   }
 
+  private static CommandsOuterClass.Commands toProto(
+      @NonNull String ledgerId,
+      @NonNull Optional<String> submissionId,
+      @NonNull CommandsSubmission submission) {
+
+    if (submission.getActAs().size() == 0) {
+      throw new IllegalArgumentException("actAs must have at least one element");
+    }
+    ArrayList<CommandsOuterClass.Command> commandsConverted =
+        new ArrayList<>(submission.getCommands().size());
```

Comment:
You don't have to provide an initial size. This is the incorrect size as well. (since it is the size of the number of `HasCommand` objects, which themselves can have many commands.

Reply:
Moreover, please use `map` instead of writing a for-loop.